### ISSUE TITLE
`gem uninstall` can't handle multiple versions of the same gem

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubygems/rubygems#6979]
Specification works fine, installation as well, but uninstallation can't handle several versions of the same gem.

## Installing
```
$ gem install simplecov:0.19.0 simplecov:0.22.0
Fetching simplecov-0.19.0.gem
Successfully installed simplecov-0.19.0
Parsing documentation for simplecov-0.19.0
Installing ri documentation for simplecov-0.19.0
Done installing documentation for simplecov after 0 seconds
Fetching simplecov-0.22.0.gem
Successfully installed simplecov-0.22.0
Parsing documentation for simplecov-0.22.0
Installing ri documentation for simplecov-0.22.0
Done installing documentation for simplecov after 0 seconds
2 gems installed
```

## Uninstalling after
```
~$ gem uninstall simplecov:0.19.0 simplecov:0.22.0
Successfully uninstalled simplecov-0.22.0
~$ gem uninstall simplecov:0.19.0 simplecov:0.22.0
Gem 'simplecov' is not installed
Successfully uninstalled simplecov-0.19.0
```

Looks like it is uniqued by name at some point

_Originally posted by @hurricup in https://github.com/rubygems/rubygems/issues/6345#issuecomment-1718745111_

**Comment by kstevens715** [comment]
If I use `gems_to_uninstall[dep.full_name]` on lines 171 and 172 in the code below, I can get the command `gem uninstall simplecov:0.19.0 simplecov:0.22.0` to work. But the tests fail because it breaks something else.

https://github.com/rubygems/rubygems/blob/7c5ba7152f504e72a9e83b47623023a25aac30c1/lib/rubygems/commands/uninstall_command.rb#L170-L179

**Comment by deivid-rodriguez** [comment]
@kstevens715 Thanks for working on this. I suggest you open a WIP PR with your proposed changes anyways, maybe we can fix it!

**Comment by kstevens715** [comment]
@deivid-rodriguez Sounds good. If I find time, I could also look into the failing tests sometime this weekend or next week.

**Comment by deivid-rodriguez** [comment]
Fixed by #7063. Thanks @kstevens715!
